inc/functions-related.php

```diff
@@ -0,0 +1,395 @@
+					$data[ $n ]['img']   = ( $image ) ? $image : get_stylesheet_directory_uri() . '/img/card-thumb.jpg';
+					$data[ $n ]['type']  = ( strpos( $data[ $n ]['url'],
+							'media.national' ) !== false ) ? 'Archives Media Player' : 'National Archives Blog';
+
+					$n ++;
+				}
+			}
+			return $data;
+		} else {
+			return false;
+		}
+	}
+}
+
+
+function display_related_content( $content, $format ) {
+
+	if ( ! is_admin() ) {
+
+		global $post;
+
+		// delete_transient( 'related-'.$format.'-'.$post->ID );
+		$transient_related = get_transient( 'related-'.$format.'-'.$post->ID );
+
+		if ( ! $transient_related ) {
+
+			$data = get_related_data( $content );
+
+			if ( $data ) {
+				$related_html     = '';
+				for ( $i = 0; $i < 3; $i ++ ) {
+					$related_html .= r_html( $post->ID, $data[ $i ]['url'], $data[ $i ]['img'], $data[ $i ]['title'],
+						$data[ $i ]['type'], $format, $i );
+				}
+			} else {
+				$related_html = false;
+			}
+
+			set_transient( 'related-'.$format.'-'.$post->ID, $related_html, DAY_IN_SECONDS );
+
+			return $related_html;
+		}
+
+		return '<!-- transient related-'.$format.'-'.$post->ID.'-->' . $transient_related . '<!-- /transient -->';
+	}
+}
+
+function blog_get_related_posts( $i = 1 ) {
+	global $post;
+	$id = $post->ID;
+
+	$tag_ids = array();
+	$tags = wp_get_post_tags($post->ID);
+	if ($tags) {
+		foreach ( $tags as $individual_tag ) {
+			$tag_ids[] = $individual_tag->term_id;
+		}
+	}
+
+	if ( is_multisite() ) {
+		$blog_list = get_sites();
+		$related = array();
+
+		if ($tag_ids) {
+			foreach ($blog_list as $blog) {
+
+				$blog_id = get_object_vars($blog)['blog_id'];
+
+				switch_to_blog( $blog_id );
+
+				$related[$blog_id] = get_posts(
+					array(
+						'tag__in' => $tag_ids,
+						'numberposts'  => $i,
+						'post__not_in' => array( $id )
+					)
+				);
+				restore_current_blog();
+			}
+		} else {
+			foreach ($blog_list as $blog) {
+
+				$blog_id = get_object_vars($blog)['blog_id'];
+
+				switch_to_blog( $blog_id );
+
+				$related[$blog_id] = get_posts(
+					array(
+						'category__in' => wp_get_post_categories($id),
+						'numberposts'  => $i,
+						'post__not_in' => array( $id )
+					)
+				);
+				restore_current_blog();
+			}
+		}
+
+	} else {
+		$related = get_posts(
+			array(
+				'tag__in' => $tag_ids,
+				'numberposts'  => $i,
+				'post__not_in' => array( $id )
+			)
+		);
+	}
+
+	return $related;
+}
+
+// [related-posts]
+function related_posts() {
+
+	global $post;
+
+	// Display posts as 'cards' or 'list'
+	$format = 'cards';
+	if (isset($_GET['ab'])) {
+		if ( ($_GET['ab']) == 'list' ) {
+			$format = 'list';
+		}
+	}
+
+	$content = display_related_content( $post->post_content, $format );
+
+	if ( $content ) {
+		if ( $format == 'cards' ) {
+			$start  = '';
+			$end    = '';
+		} else {
+			$start   = '<ul class="documents">';
+			$end     = '</ul>';
+		}
+
+		$html  = '<div class="related-posts related-post-thumbs clearfix">';
```

Comment:
Minor stylistic point but I'd probably opt to use a single approach for building up HTML content within the PHP, either the concatenation here or the interpolation above.

Reply:
Yeah, I think I was lazy. I copied this from another theme template. But if we go forward with this prototype I'll rebuild this from scratch as I feel there are too many moving parts. I think the logic overall can be simplified.